datafusion/sqllogictest/test_files/array.slt

```diff
@@ -1752,6 +1752,34 @@ select array_to_string(make_array(), ',')
 ----
 (empty)
 
+
+## array_union (aliases: `list_union`)
+
+# array_union scalar function #1
+query ?
+select array_union([1, 2, 3, 4], [5, 6, 3, 4]);
+----
+[1, 2, 3, 4, 5, 6]
+
+# array_union scalar function #2
+query ?
+select array_union([1, 2, 3, 4], [5, 6, 7, 8]);
+----
+[1, 2, 3, 4, 5, 6, 7, 8]
+
+# array_union scalar function #3
+query ?
+select array_union([1,2,3], []);
+----
+[1, 2, 3]
+
+# array_union scalar function #4
+query ?
+select array_union([1, 2, 3, 4], [5, 4]);
+----
+[1, 2, 3, 4, 5]
```

Comment:
Your implementation cover the column-wise cases so it is better to add them into test. Also, you can also float type, I had test it, your code can pass them too.

Reply:
What's a column wise case? Something like this?

```
select array_union([[1, 2, 3, 4], [5, 4]], [[1, 2, 3, 4], [5, 4]]);
```

Reply:
```text
statement ok
CREATE TABLE arrays_with_repeating_elements
AS VALUES
  ([1], [2]),
  ([2, 3], [3]),
  ([3], [3, 4])
;

query ?
select array_union(column1, column2) from arrays_with_repeating_elements;
----
[1, 2]
[2, 3]
[3, 4]

statement ok
drop table arrays_with_repeating_elements;
```

Reply:
Fixed with https://github.com/apache/arrow-datafusion/commit/648ae5d659142082d085418dc7b84ad857ba8136